src/ERC1155Common.sol

```diff
@@ -1,80 +1,149 @@
-// SPDX-License-Identifier: UNLICENSED
-pragma solidity ^0.8.0;
+// SPDX-License-Identifier: MIT
+// Compatible with OpenZeppelin Contracts ^5.0.0
+pragma solidity ^0.8.20;
 
+import { AccessControlEnumerable } from "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
 import { ERC1155 } from "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+
+import { ERC1155Burnable } from "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Burnable.sol";
+import { ERC1155Pausable } from "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Pausable.sol";
 import { ERC1155Supply } from "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Supply.sol";
-import { AccessControlEnumerable } from "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
 import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
 import { IERC1155Common } from "./interfaces/IERC1155Common.sol";
 
-contract ERC1155Common is AccessControlEnumerable, ERC1155, ERC1155Supply, IERC1155Common {
+contract ERC1155Common is
+  ERC1155,
+  AccessControlEnumerable,
+  ERC1155Pausable,
+  ERC1155Burnable,
+  ERC1155Supply,
+  IERC1155Common
+{
   using Strings for uint256;
 
+  bytes32 public constant URI_SETTER_ROLE = keccak256("URI_SETTER_ROLE");
+  bytes32 public constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
   bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
+
   string private _name;
   string private _symbol;
 
-  constructor(string memory name_, string memory symbol_, string memory baseTokenURI, address[] memory minters)
-    payable
-    ERC1155(baseTokenURI)
-  {
+  constructor(address admin, string memory name_, string memory symbol_, string memory uri_) ERC1155(uri_) {
+    _grantRole(DEFAULT_ADMIN_ROLE, admin);
+    _grantRole(PAUSER_ROLE, admin);
+    _grantRole(MINTER_ROLE, admin);
+    _grantRole(URI_SETTER_ROLE, admin);
+
```

Comment:
should we have control parameter in case we dont want our nfts to be burned by user?

Reply:
Let the partner handle this based on their needs